source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/TextBlockRenderer.java

```diff
@@ -68,7 +71,25 @@ public static void setTextSize(TextView textView, FontType type, TextSize textSi
 
     public void setTextFormat(TextView textView, HostConfig hostConfig, FontType type, TextWeight textWeight)
     {
-        textView.setTypeface(TextRendererUtil.getTextFormat(hostConfig, type), m_textWeightMap.get(textWeight));
+        Typeface typeface = TextRendererUtil.getTextFormat(hostConfig, type, textWeight == TextWeight.Lighter);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P)
+        {
+            Typeface weightedTypeface = Typeface.create(typeface, (int) hostConfig.GetFontWeight(type, textWeight), false);
+            textView.setTypeface(weightedTypeface);
+        }
+        else
+        {
```

Comment:
be nice to add some sort of comments here as to why the build version check is needed.